news/_posts/2022-01-05-multistore-configuration-forms.md

```diff
@@ -0,0 +1,69 @@
+
+One of the back-office aspects of this feature is that many configuration forms can have different values for different shops, which is dealt with using shop contexts.
+
+## How it works and what was less than stellar about it
+
+In a configuration form, a merchant can set different values for different shop contexts, of which there are three, in this order of inheritence:
+
+- All shop context
+- Group shop context
+- SIngle shop context
+
+![Shop contexts](/assets/images/2022/01/shopcontexts.png)
+
+When a value is not set for a context, it inherits the value of the parent context. Consequently, the "All Shops" context values can be considered a default values. 
+
+Unfortunately, before Prestashop 1.7.8, this mechanism was not obvious on the forms in question, and it was not easy to see at a glance which form fields were modified or not for the current context.
+
+Something had to be done in order to make this feature more understandable.
+
+## Objectives
+
+Our multistore implementation had to meet 2 main criterias:
+
+- Improve existing UX for better understanding
+
+- Make the technical implementation as easily reusable as possible on existing and future configuration forms, so external contributors and module developers can easily reuse our implementation in configuration forms.
+
+## How we did it
+
+### The UX
+
+After numerous merchant interviews, our UX team created mock-ups, then our product team wrote functional specifications in accordance with merchant feedback. 
+
+Here is how a multistore compatible configuration form looks like since Prestashop 1.7.8:
+
+IMAGE => capture ecran multistore
+
+As you can see, a checkbox appears before multistore compatible fields, when it's not checked and the field is greyed out, it means that the value is inherited from a prent context. When it's checked, the value is overridden. 
+On the right of the field, a dropdown tells you if the value is overridden or not for each context.
+
+### The technical implementation
+
+On the technical side, these visual elements had to be reusable and easily implementable, without having to copy HTML or rethink the logic in the backend. 
+
+All the work of knowing if the value of a field is inherited or not, and saving the submitted value only for the right context should be done only once, and easily reused by external contributors.
+
+For this purpose we decided to take advantage of Symfony form extensions, so that to add the new checkboxes and multistore dropdowns on a configuration form, a developer only needs to extend our our multistore extension. 
+
+Then we added multistore helper methods for Configuration classes, to make saving data for the right context depending on which checkboxes are checked a breeze. 
+
+Finally, we created a new multistore javascript component, which only needs to be instantiated in order to make the multistore form reactive.
```

Comment:
Link to PR can be 👍

Reply:
There's also a dedicated module to show the multi store configuration form, right @matthieu-rolland ?